zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+if(CONFIG_BOOTLOADER_MCUBOOT)
```

Comment:
Might want to drop a README in this directory or add a comment to this file saying that the `zephyr` directory is for use of mcuboot as a zephyr module (https://docs.zephyrproject.org/latest/application/index.html#modules-external-projects), and that boot/zephyr is where the Zephyr-specific application code is, to avoid confusing newcomers who don't know to look for boot/zephyr yet.

I've had to clear up confusion that CONFIG_BOOTLOADER_MCUBOOT should *not* be enabled when building mcuboot itself, but is rather an option that's enabled by applications using MCUboot; this file muddies that distinction a bit, so a comment would be nice to clear up that what's going on here is kind of complex:

1. Application CMakeLists.txt
2. Application boilerplate.cmake
3. Application zephyr_modules.cmake
4. This file
5. MCUboot CMakeLists.txt (boot/zephyr/CMakeLists.txt)
6. MCUboot boilerplate.cmake
...